From 0f10601a81442c2e2379b1c5e8a855e085e4febf Mon Sep 17 00:00:00 2001
From: Evan Wang <xswang@marvell.com>
Date: Thu, 20 Jul 2017 10:04:43 +0800
Subject: [PATCH 1196/1345] dts: cp110: force xhci reset after resume

commit  9154dabd1d84dc040a03e4acbaa03879a6978d05 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Some SOCs such as CP110, armada 37xx, must reset XHCI
after resume.
Armada-CP110 & Armada-37xx share the same XHCI IP,
and reset-on resume flag enabled in Armada-37xx.

This patch set the property of "needs-reset-on-resume"
in CP110 USB DT node. When this property is set in DT,
xhci quirks will set XHCI_RESET_ON_RESUME flag, and the
XHCI resume will reset the PHY.

Change-Id: I4716314dc459779cf1e909fdf4cf5c4130970d21
Signed-off-by: Evan Wang <xswang@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/42896
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi |    2 ++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi   |    2 ++
 2 files changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
index 0831c18..d104550 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
@@ -91,6 +91,7 @@ usb3h0_1: usb3@500000 {
 	dma-coherent;
 	interrupts = <ICU_GRP_NSR 106 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cps_syscon0 1 22>;
+	needs-reset-on-resume;
 	status = "disabled";
 };
 
@@ -100,6 +101,7 @@ usb3h1_1: usb3@510000 {
 	dma-coherent;
 	interrupts = <ICU_GRP_NSR 105 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cps_syscon0 1 23>;
+	needs-reset-on-resume;
 	status = "disabled";
 };
 
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 42e800a..8729b28 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -174,6 +174,7 @@ usb3h0: usb3@500000 {
 	dma-coherent;
 	interrupts = <ICU_GRP_NSR 106 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cpm_syscon0 1 22>;
+	needs-reset-on-resume;
 	status = "disabled";
 };
 
@@ -183,6 +184,7 @@ usb3h1: usb3@510000 {
 	dma-coherent;
 	interrupts = <ICU_GRP_NSR 105 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cpm_syscon0 1 23>;
+	needs-reset-on-resume;
 	status = "disabled";
 };
 
-- 
1.7.9.5

